src/optimizer/query_rewrite.c

```diff
@@ -3074,6 +3088,410 @@ qo_reduce_comp_pair_terms (PARSER_CONTEXT * parser, PT_NODE ** wherep)
+ *     only have one of my spec   '='    not have my spec node
+ * e.g.)      col(my spec)         =     2 (constant)
+ *            a.col(my spec)       =     b.col
+ *            a.col(my spec)       =     b.col + c.col
+ */
+static PT_NODE *
+qo_collect_name_with_eq_const (PARSER_CONTEXT * parser, PT_NODE * on_cond, PT_NODE * spec)
+{
+  PT_NODE *pred, *arg1, *arg2;
+  PT_NODE *point, *s_name, *point_list = NULL;
+  SPEC_CNT_INFO info1, info2;
+
+  info1.spec = spec;
+  info2.spec = spec;
+
+  pred = on_cond;
+  while (pred != NULL)
+    {
+      if (pred->or_next != NULL)
+	{
+	  pred = pred->next;
+	  continue;
+	}
+      if (!PT_IS_EXPR_NODE_WITH_OPERATOR (pred, PT_EQ))
+	{
+	  pred = pred->next;
+	  continue;
+	}
+      /* check on_cond predicate */
+      if (spec->info.spec.location != pred->info.expr.location)
+	{
+	  pred = pred->next;
+	  continue;
+	}
+
+      arg1 = pred->info.expr.arg1;
+      arg2 = pred->info.expr.arg2;
+      /* find name with spec id */
+      info1.my_spec_cnt = 0;
+      info1.other_spec_cnt = 0;
+      info2.my_spec_cnt = 0;
+      info2.other_spec_cnt = 0;
+      parser_walk_tree (parser, arg1, qo_get_name_cnt_by_spec, &info1, NULL, NULL);
+      parser_walk_tree (parser, arg2, qo_get_name_cnt_by_spec, &info2, NULL, NULL);
+
+      /* col1(my spec) = const */
+      if (info1.my_spec_cnt == 1 && info1.other_spec_cnt == 0)
+	{
+	  /* const */
+	  if (info2.my_spec_cnt == 0)
+	    {
+	      point_list = parser_append_node (pt_point (parser, info1.my_spec_node), point_list);
+	    }
+	}
+      /* const = col(my spec) */
+      else if (info2.my_spec_cnt == 1 && info2.other_spec_cnt == 0)
+	{
+	  /* const */
+	  if (info1.my_spec_cnt == 0)
+	    {
+	      point_list = parser_append_node (pt_point (parser, info2.my_spec_node), point_list);
+	    }
+	}
+      pred = pred->next;
+    }
+  return point_list;
+}
+
+/*
+ * qo_modify_location () -
+ *   return:
+ *   parser(in):
+ *   node(in):
+ *   arg(in):
+ *   continue_walk(in):
+ */
+static PT_NODE *
+qo_modify_location (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  RESET_LOCATION_INFO *infop = (RESET_LOCATION_INFO *) arg;
+
+  if (node->node_type == PT_EXPR && node->info.expr.location == infop->start)
+    {
+      node->info.expr.location = infop->end;
+    }
+
+  if (node->node_type == PT_NAME && node->info.name.location == infop->start)
+    {
+      node->info.name.location = infop->end;
+    }
+
+  if (node->node_type == PT_VALUE && node->info.value.location == infop->start)
+    {
+      node->info.value.location = infop->end;
+    }
+
+  return node;
+}
+
+/*
+ * qo_reset_spec_location () - reset location of spec
+ *   return: node_list
+ *   parser(in):
+ *   on_cond(in):
+ *   spec(in):
+ *
+ * Note:
+ */
+static void
+qo_reset_spec_location (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE * query)
+{
+  short curr_loc, after_loc;
+  PT_NODE *where;
+  RESET_LOCATION_INFO locate_info;
+
```

Comment:
```suggestion
PT_NODE *sp

sp = spec;
while (sp != NULL)
    {
      ...
      sp = sp->next;
    }
```

Reply:
Since the spec is only used here, I think it would be ok to keep it.